log4j-api/src/main/java/org/apache/logging/log4j/status/StatusLogger.java

```diff
@@ -16,311 +16,590 @@
+            return debug != null;
+        }
+
+        private static int readBufferCapacity(final Properties fileProvidedProperties) {
+            final String capacityString = readProperty(fileProvidedProperties, BUFFER_CAPACITY_PROPERTY_NAME);
+            return capacityString != null ? Integer.parseInt(capacityString) : BUFFER_CAPACITY_DEFAULT_VALUE;
+        }
 
-    private static final String NOT_AVAIL = "?";
+        private static Level readDefaultListenerLevel(final Properties fileProvidedProperties) {
+            final String level = readProperty(fileProvidedProperties, DEFAULT_LISTENER_LEVEL_PROPERTY_NAME);
+            return level != null ? Level.valueOf(level) : DEFAULT_LISTENER_LEVEL_DEFAULT_VALUE;
+        }
 
-    static final PropertiesUtil PROPS = new PropertiesUtil("log4j2.StatusLogger.properties");
+        private static DateTimeFormatter readInstantFormatter(final Properties fileProvidedProperties) {
+            final String format = readProperty(fileProvidedProperties, INSTANT_FORMAT_PROPERTY_NAME);
+            return format != null ? DateTimeFormatter.ofPattern(format) : null;
+        }
 
-    private static final int MAX_ENTRIES = PROPS.getIntegerProperty(MAX_STATUS_ENTRIES, 200);
+        private static String readProperty(final Properties fileProvidedProperties, final String propertyName) {
+            final String systemProvidedValue = System.getProperty(propertyName);
+            return systemProvidedValue != null
+                    ? systemProvidedValue
+                    : (String) fileProvidedProperties.get(propertyName);
+        }
 
-    private static final String DEFAULT_STATUS_LEVEL = PROPS.getStringProperty(DEFAULT_STATUS_LISTENER_LEVEL);
+        // We need to roll out our own `.properties` reader.
+        // We could have used `PropertiesUtil`, `PropertyFilePropertySource`, etc.
+        // Consequently, they would delegate to `LoaderUtil`, etc.
+        // All these mechanisms expect a working `StatusLogger`.
+        // Hence, in order to be self-sufficient, we cannot rely on them.
+        private static Properties readPropertiesFile() {
+            final Properties properties = new Properties();
+            final URL url = StatusLogger.class.getResource(PROPERTIES_FILE_NAME);
+            if (url == null) {
+                return properties;
+            }
+            try (final InputStream stream = url.openStream()) {
+                properties.load(stream);
+            } catch (final IOException error) {
+                // There is no logging system at this stage.
+                // There is nothing we can do but simply dumping the failure.
+                error.printStackTrace(System.err);
+            }
+            return properties;
+        }
+    }
 
-    static final boolean DEBUG_ENABLED =
-            PropertiesUtil.getProperties().getBooleanProperty(Constants.LOG4J2_DEBUG, false, true);
+    private static volatile StatusLogger INSTANCE = new StatusLogger();
```

Comment:
Note that by putting this in a static field, this means the static init of this class will cause the construction of the field. You could wrap it in a `Lazy` to defer that until something calls `getLogger` if you'd like.

Reply:
Good catch. Fixed in 3684a2e2de6132f2d367e3887f164d0626878a27.